src/website/interface.ts

```diff
@@ -102,6 +102,15 @@ export class WebsiteInterface {
 		}
 	}
 
+	public async findClosedMatch(tournamentId: string, playerId: number): Promise<WebsiteMatch | undefined> {
+		const currentRound = await this.getRound(tournamentId);
+		const playerMatches = await this.api.getMatches(tournamentId, false, playerId);
```

Comment:
You're calling getMatches twice in the same function here.

Reply:
You mean in submitScoreForce? Because I'm not directly in this function, and the idea is to try to find an open match first. ...then again it will be either open or closed for sure so you're probably right that we don't need to, it can find the open match in the list of closed matches. Thanks for the catch

Reply:
This is another case of an abstraction very much obscuring what's going on. Here you can see the (rather trivial) implementation of startRound below, which calls getMatches, but it happens immediately again here, and this means that you have two round trips to the same endpoint.

Reply:
OK, I think my confusion here stems from this being a line comment. I can understand that I'm likely calling getMatches` twice in some `TournamentManager` process, and I'm on board with changing things so that I'm no longer doing so. But I don't think `findClosedMatch`, which is used only for `forcescore`, has anything to do with that. Did you leave this comment on the wrong line by mistake?

Reply:
I made a typo. `this.getRound` literally calls getMatches, right below 👀